app/scripts/app.js

```diff
@@ -15,6 +15,18 @@ blocChat.config(['$stateProvider', '$locationProvider', function($stateProvider,
     });
 }]);
 
-blocChat.controller('Home.controller', ['$scope', function($scope) {
+blocChat.factory('Room', ['$firebaseArray', function($firebaseArray) {
+    var roomsUrl = 'https://stewart-bloc-chat.firebaseio.com/rooms';
+    var roomsRef = new Firebase(roomsUrl);
```

Comment:
Try thinking about it from the other side, from the client of your factory. How would you expect it to work? How would you expect to be written, in code?

If `Room.all` sounds weird compared to something like `Room.rooms` or `Room.objects`, then you might be better off changing it.

As for your actual factory, it's a bit odd to use it as an argument to anything, let alone to `$firebaseArray` -- which is why none of my example usages above work in your controller.